api/shared/interface.go

```diff
@@ -45,3 +56,51 @@ type GroupHandler interface {
 	)
 	IsInterfaceNil() bool
 }
+
+// FacadeHandler defines all the methods that a facade should implement
+type FacadeHandler interface {
+	GetBalance(address string) (*big.Int, error)
+	GetUsername(address string) (string, error)
+	GetValueForKey(address string, key string) (string, error)
+	GetAccount(address string) (api.AccountResponse, error)
+	GetESDTData(address string, key string, nonce uint64) (*esdt.ESDigitalToken, error)
+	GetESDTsRoles(address string) (map[string][]string, error)
+	GetNFTTokenIDsRegisteredByAddress(address string) ([]string, error)
+	GetESDTsWithRole(address string, role string) ([]string, error)
+	GetAllESDTTokens(address string) (map[string]*esdt.ESDigitalToken, error)
+	GetKeyValuePairs(address string) (map[string]string, error)
+	GetBlockByHash(hash string, withTxs bool) (*api.Block, error)
+	GetBlockByNonce(nonce uint64, withTxs bool) (*api.Block, error)
+	Trigger(epoch uint32, withEarlyEndOfEpoch bool) error
+	IsSelfTrigger() bool
+	GetTotalStakedValue() (*api.StakeValues, error)
+	GetDirectStakedList() ([]*api.DirectStakedValue, error)
+	GetDelegatorsList() ([]*api.Delegator, error)
+	StatusMetrics() external.StatusMetricsHandler
+	GetAllIssuedESDTs(tokenType string) ([]string, error)
+	GetHeartbeats() ([]data.PubKeyHeartbeat, error)
+	GetQueryHandler(name string) (debug.QueryHandler, error)
+	GetPeerInfo(pid string) ([]core.QueryP2PPeerInfo, error)
+	GetNumCheckpointsFromAccountState() uint32
+	GetNumCheckpointsFromPeerState() uint32
+	GetProof(rootHash string, address string) ([][]byte, error)
+	GetProofCurrentRootHash(address string) ([][]byte, []byte, error)
+	VerifyProof(rootHash string, address string, proof [][]byte) (bool, error)
+	GetThrottlerForEndpoint(endpoint string) (core.Throttler, bool)
+	CreateTransaction(nonce uint64, value string, receiver string, receiverUsername []byte, sender string, senderUsername []byte, gasPrice uint64,
```

Comment:
Maybe use a new `args` struct instead.

Reply:
I would leave it as it is, because most fields are primitives, and default values can easily appear

Reply:
Ok.